File: 연습장/Python/update.py
#!python
print("Content-type: text/html; charset=utf-8\n")
# ↑ Python을 연동시켜서 화면에 도출될 수 있게 하는 코드 ↑ (1~2)

import cgi, os #os라는 모듈로 묶여서 실행되는 명령어를 실행시킬 수 있음

form = cgi.FieldStorage()
if "id" in form: #form안에 id 값이 있는 경우 else에 해당하는 값을 도출
    pageId = form["id"].value
    description = open("Data/"+pageId, "r").read()
else:
    pageId = "Welcome"
    description = "Hello, Web"

files = os.listdir("Data") #files라는 함수로 묶어서 Data 폴더에 있는 파일을 불러온다

# for data in files:
#     listStr = listStr + data #비어있는 문자열 + Data 내의 파일
# print(listStr) #공백 + CSS + HTML + JavaScript + Python

listStr = "" #비어있는 문자열 임의로 생성

for item in files:
    listStr = listStr + '<li><a href="index.py?id={name}">{name}</a></li>'.format(name=item)
#''작은 따옴표로 <li></li>를 묶어야 하나 봄
# li 형식으로 item이라는 함수를 name으로 포맷팅을 한 후, 웹 주소도 id 값을 {name}으로 포맷팅,
# li의 이름도 {name}으로 포맷팅해서 화면에 도출

print('''<!doctype html>
<html>
<head>
  <title>WEB1 - Welcome</title>
  <meta charset="utf-8">
</head>
<body>
  <h1><a href="index.py">WEB</a></h1>
  <ol>
    {listStr} <!-- 위에서 만든 파일 불러오기 포맷팅을 이식 -->
  </ol>
  <a href="create.py">Create</a>
  <form action="process_update.py" method="post">
  <!-- 전송할 파일을 action 태그로 지정, method를 post로 지정해서 함부로 수정할 수 없도록 하기 -->
      <input type="hidden" name="pageId" value="{form_default_title}"></p>
      <!--제목이 아닌 pageId 값의 수정을 막기 위해 pageId 수정 칸을 따로 만들고 숨기기 -->
      <p><input type="text" name="Title" placeholder="Title" value="{form_default_title}"></p>
      <!-- 제목 생성을 위한 input 태그 / 수정 시에 텍스트 박스에 원래 글이 남아있도록 value 값을 지정 -->
      <p><textarea rows="5" name="Description" placeholder="Description">{form_default_desc}</textarea></p>
      <!-- 여러줄의 내용 입력을 위한 textbox / 수정 시에 텍스트 박스에 원래 글이 남아있도록 태그 사이에 값 지정 -->
      <p><input type="submit"></p> <!--  제출 버튼 -->
  </form> <!-- 제목과 내용을 제출하기 위해 form으로 감싸기 -->
</body>
</html>
'''.format(title=pageId, desc=description, listStr=listStr, form_default_title=pageId, form_default_desc=description))
# 수정사항의 경우 form_default_title에는 pageId, form_default_desc에는 description을 지정

# ?id=xxx -> 쿼리스트링

File: 연습장/PHP/update.php
<?php
  require_once("lib/print.php");
  // top.php에 겹치는 require을 한번만 실행하게끔 지정
  require("view/top.php");
  // print.php 파일의 코드를 읽어오도록 하는 코드
?>
    <a href="create.php">생성</a>
    <?php if(isset($_GET["id"])) { ?>
      <!-- isset을 통해 id값의 유무에 따른 true, false 확인 후
      id가 없을 시 수정 버튼이 보이지 않도록 지정-->
      <a href="update.php?id=<?=$_GET["id"]?>">수정</a>
      <!-- 수정을 누를 시 원래 있던 제목과 본문이 글 박스에 남아있도록 지정 -->
      <!-- <?=$_GET["id"]?>와 <?php echo $_GET["id"]; ?>은 같음 -->
    <?php } ?>
  <h2>
    <?php
      print_title(); // 위에서 지정한 제목 함수 출력
    ?>
  </h2>
    <?php
      print_description(); // 위에서 지정한 목록 함수 출력
    ?>
    <form action="update_process.php" method="post">
      <!-- 누군가에게 링크를 보낼때 get 방식의 경우 파일이 즉석에서 생성되기 때문에
      방지를 위해 post 방식으로 사용, _process.php에도 post 형식으로 id값을 불러와야 함 -->
      <input type="hidden" name="old_title" value="<?=$_GET["id"]?>">
      <!-- 원래의 파일명을 input 숨김을 통해 서버로 원래의 파일명 전송 -->
      <!-- 숨겨진 input에 <?=$_GET["id"]?>를 통한 원래 파일명을 지정 -->
      <p>
        <input type="text" name="title" placeholder="제목을 입력하세요."
        value="<?php print_title(); ?>">
        <!-- value : 글 박스에 남아있는 글씨 지정
        print_xxx();을 통해 지정한 제목을 value값으로 남겨두기-->
      </p>
      <p>
        <textarea name="description" placeholder="본문을 입력하세요."
        rows="8" cols="30">"<?php print_description(); ?>"</textarea>
        <!-- textarea 태그 사이에 본문에 해당하는 php코드 입력 -->
      </p>
      <p>
        <input type="submit">
      </p>
    </form>
<?php
  require("view/bottom.php");
?>
